www/class/centreonEscaping.php

```diff
@@ -0,0 +1,171 @@
+     *
+     * Usage:
+     * $sMyHtmlContent = '<div>'.Esc::forHtmlContent($sMyContent).' <span>'.Esc::forHtmlContent($sMySubContent).'</span></div>';
+     *
+     * @param $sValue string 
+     * @return string The escaped string, ready for injection in HTML content 
+     *    (ie. BETWEEN elements!)
+	 */
+	static public function forHtmlContent($sValue) {
+		$sCharSet = ini_get("default_charset") ? ini_get("default_charset") : "UTF-8";
+		return htmlspecialchars($sValue, ENT_QUOTES, $sCharSet, true);
+	}
+
+	/**
+     * Escape for Javascript values
+     * 
+     * For example, in case for:
+     * <script type="text/javascript"> var hostname='[INJECT UNTRUSTED STRING HERE]'; </script>
+     *
+     * Usage:
+     * <script type="text/javascript"> var hostname="<?php echo Esc::forJsValue($sMyHostname); ?>"; </script>
+     *
+     * @param $sValue string 
+     * @param $bWithQuotes bool Return the value with encapsulated 
+     *    quotes (default: false, which is consistent with the other functions)
+     * @return string The escaped string, ready for injection in Javascript 
+     *    values (ie. BETWEEN quotes when $bWithQuotes is set to false!)
+	 */
+	static public function forJsValue($sValue, $bWithQuotes = false) {
+		//We reuse the power of json_encode to encode strings for Javascript. 
+		//The output is normally already encapsulated with quotes, which we 
+		//may need to remove depending on usage.
+		//Note: json_encode expects the string to be UTF8
+		$sValue = json_encode((string)$sValue, JSON_HEX_QUOT | JSON_HEX_APOS);
+		return $bWithQuotes ? $sValue : mb_substr($sValue, 1, -1);
+	}
+	
+	
+
+	/**
+	 * Escape for Database Values. Alias for: CentreonDB::escape() (preferred, 
+	 *   as it is used everywhere already). The function has been created here for completeness. 
+	 *
+	 * For example, in case for:
+	 * SELECT * FROM hosts WHERE host_name='[INJECT UNTRUSTED STRING HERE]' OR host_alias LIKE '%[INJECT UNTRUSTED STRING HERE]%'
+	 *
+	 * Usage:
+	 * $sMyHostsQuery = "SELECT * FROM hosts WHERE host_name='".Esc::forQueryValue($sMyHostName)."' OR host_alias LIKE '%".Esc::forQueryValue($sMyHostName)."%'";
+	 *
+	 * @param $sValue string
+	 * @return string The escaped string, ready for injection in query
+	 *   values (ie. BETWEEN quotes!)
+	 */
+	static public function forQueryValue($sValue) {
+		return CentreonDB::escape($sValue);
+	}
+}
+
+/*
+ * A class for convenience. 
+ * This class makes short code style possible using Esc::forHtmlAttr(). 
+ * 
+ * This is prefered as opposed to CentreonEscape::forHtmlAttr(), which 
+ * can be long and unclear, unoverseeable in lines with more than 3 values to 
+ * escape, and hence prone to security mistakes. This class makes it clearer 
+ * for developers' understanding and analogous with verbal understanding 
+ * eg. "Escape (this value) for Html Attributes":  Esc::forHtmlAttr($sThisValue)
+ */
+class Esc extends CentreonEscape { }
```

Comment:
hi, can you make a file by class, and name the file like the class.

Reply:
See comment below, the request has been done